clustermesh/clustermesh.go

```diff
@@ -1885,3 +1941,199 @@ func DisableWithHelm(ctx context.Context, k8sClient *k8s.Client, params Paramete
+	version := localRelease.Chart.AppVersion()
+	semv, err := utils.ParseCiliumVersion(version)
+	if err != nil {
+		return fmt.Errorf("failed to parse Cilium version: %w", err)
+	}
+
+	v := fmt.Sprintf("%d.%d.%d", semv.Major, semv.Minor, semv.Patch)
+	k.Log("✅ Detected Helm release with Cilium version %s", v)
+
+	if v < "1.14.0" {
+		// Helm-based clustermesh enable is only supported on Cilium v1.14+ due to a lack of support in earlier versions
+		// for autoconfigured certificates (tls.{crt,key}) for cluster members when running in certgen (cronJob) PKI
+		// mode
+		k.Log("⚠️ Cilium Version is less than 1.14.0. Continuing in classic mode.")
+		return k.Connect(ctx)
+	}
+
+	localClient, remoteClient, err := k.getClientsForConnect()
+	if err != nil {
+		return err
+	}
+
+	aiLocal, aiRemote, err := k.getAccessInfoForConnect(ctx, localClient, remoteClient)
+	if err != nil {
+		return err
+	}
+
+	err = k.validateInfoForConnect(aiLocal, aiRemote)
+	if err != nil {
+		return err
+	}
+
+	// Validate that CA certificates match between the two clusters
+	err = k.validateCAMatch(aiLocal, aiRemote)
+	if err != nil {
+		return err
+	}
+
+	// Get existing helm values for the local cluster
+	localHelmValues := localRelease.Config
+	// Expand those values to include the clustermesh configuration
+	localHelmValues, err = updateClustermeshConfig(localHelmValues, aiLocal, aiRemote)
+	if err != nil {
+		return err
+	}
+
+	// Get existing helm values for the remote cluster
+	remoteRelease, err := getRelease(remoteClient, k.params.Namespace)
+	if err != nil {
+		k.Log("❌ Unable to find Helm release for the remote cluster")
+		return err
+	}
+	remoteHelmValues := remoteRelease.Config
+	// Expand those values to include the clustermesh configuration
+	remoteHelmValues, err = updateClustermeshConfig(remoteHelmValues, aiRemote, aiLocal)
+	if err != nil {
+		return err
+	}
+
+	upgradeParams := helm.UpgradeParameters{
+		Namespace:   k.params.Namespace,
+		Name:        defaults.HelmReleaseName,
+		Values:      localHelmValues,
+		ResetValues: false,
+		ReuseValues: true,
+	}
+
+	// Enable clustermesh using a Helm Upgrade command against our target cluster
+	k.Log("⚠️ Configuring Cilium in cluster '%s' to connect to cluster '%s'",
+		localClient.ClusterName(), remoteClient.ClusterName())
+	_, err = helm.Upgrade(ctx, localClient.RESTClientGetter, upgradeParams)
+	if err != nil {
+		return err
+	}
+
+	// Enable clustermesh using a Helm Upgrade command against the remote cluster
+	k.Log("⚠️ Configuring Cilium in cluster '%s' to connect to cluster '%s'",
+		remoteClient.ClusterName(), localClient.ClusterName())
+	upgradeParams.Values = remoteHelmValues
+	_, err = helm.Upgrade(ctx, remoteClient.RESTClientGetter, upgradeParams)
+	if err != nil {
+		return err
+	}
+
+	k.Log("✅ Connected cluster %s and %s!", localClient.ClusterName(), remoteClient.ClusterName())
+	return nil
+}
+
+func updateClustermeshConfig(
+	values map[string]interface{},
+	aiLocal, aiRemote *accessInformation,
```

Comment:
Given that `updateClustermeshConfig` needs to be called twice (once for the local cluster and once for the remote cluster), you could drop one of the `accessInformation` parameters, and pass only the appropriate one.

Reply:
`updateClustermeshConfig` is being called once against the Helm values of the local cluster, and once against the Helm values of the remote cluster, each time it needs to ensure that both the information from `aiLocal` and `aiRemote` is contained in those Helm values.

Reply:
In the config, you need only to specify the parameters about the remote clusters, while the ones for the local cluster are not necessary (and they are indeed ignored if present).

Reply:
I see, that's surprising to me, and appears to be under-documented.